src/org/traccar/events/GeofenceEventHandler.java

```diff
@@ -0,0 +1,74 @@
+package org.traccar.events;
+
+import java.sql.SQLException;
+import java.util.Set;
+
+import org.traccar.BaseEventHandler;
+import org.traccar.Context;
+import org.traccar.database.DataManager;
+import org.traccar.database.GeofenceManager;
+import org.traccar.helper.Log;
+import org.traccar.model.Device;
+import org.traccar.model.Event;
+import org.traccar.model.Position;
+
+public class GeofenceEventHandler extends BaseEventHandler {
+
+    private int suppressRepeated;
+    private GeofenceManager geofenceManager;
+    private DataManager dataManager;
+
+    public GeofenceEventHandler() {
+        suppressRepeated = Context.getConfig().getInteger("event.suppressRepeated", 60);
+        geofenceManager = Context.getGeofenceManager();
+        dataManager = Context.getDataManager();
+    }
+
+    @Override
+    protected Event analizePosition(Position position) {
+        Event event = null;
+        if (!isLastPosition() || !position.getValid()) {
+            return event;
+        }
+
+        Device device = dataManager.getDeviceById(position.getDeviceId());
+        if (device == null) {
+            return event;
+        }
+
+        Set<Long> geofences = geofenceManager.getAllDeviceGeofences(position.getDeviceId());
+        if (geofences == null) {
```

Comment:
I agree that it should be possible to have multiple events. I'm guessing that some users would want to draw geofences next to each other and it's likely that object leaves one geofence and enters another one straight away.

Also, it should be possible to draw geofences on top of each other. It means that each object can be in multiple geofences at the same time.